cylc/flow/task_pool.py

```diff
@@ -1332,7 +1382,14 @@ def sim_time_check(self, message_queue):
                 sim_task_state_changed = True
         return sim_task_state_changed
 
-    def set_expired_task(self, itask, now):
+    def set_expired_tasks(self):
+        res = False
+        for itask in self.get_tasks():
+            if self._set_expired_task(itask):
+                res = True
+        return res
```

Comment:
```suggestion
        for itask in self.get_tasks():
            if self._set_expired_task(itask):
                return True
        return False
```

Reply:
Ooh, damn, I didn't spot that one.

Reply:
Yikes, I apologize to every coder in the world for this abomination :face_with_head_bandage:

Reply:
Wait, my bad, didn't think about the side effects of `self._set_expired_task()`